packages/elvis-dropdown-lit/src/e-dropdown.ts

```diff
@@ -0,0 +1,169 @@
+
+@customElement('e-dropdown')
+export class EDropdown extends LitElement {
+  static styles = css`
+    :host {
+      --width: 448px;
+      --option-height: 48px;
+      display: block;
+      width: var(--width);
+      position: relative;
+    }
+
+    .input {
+      width: inherit;
+      border: 1px solid ${unsafeCSS(getThemeColor('border-1'))};
+      border-radius: 8px;
+
+      display: flex;
+      justify-content: space-between;
+      align-items: center;
+      gap: 8px;
+      padding: 8px 8px 8px 16px;
+
+      input {
+        all: unset;
+        width: inherit;
+      }
+
+      &:focus-within {
+        border-color: ${unsafeCSS(getThemeColor('border-hover-1'))};
+      }
+
+      .icon {
+        display: grid;
+        place-items: center;
+
+        transition: transform 0.2s ease-in-out;
+        &.open {
+          transform: rotate(180deg);
+        }
+      }
+    }
+
+    .options {
+      position: absolute;
+      top: calc(100% + 8px);
+      left: 0;
+      width: calc(var(--width) - calc(2 * 8px));
+      display: flex;
+      flex-direction: column;
+      box-shadow: ${unsafeCSS(getShadow('medium'))};
+      border-radius: 8px;
+      z-index: 1;
+      max-height: calc(var(--option-height) * 5.5);
+      overflow-y: auto;
+      background-color: ${unsafeCSS(getThemeColor('background-overlay-1'))};
+    }
+
+    .options e-option {
+      &:hover {
+        z-index: 2;
+        background-color: ${unsafeCSS(getThemeColor('background-hover-1'))};
+      }
+    }
+  `;
+
+  @property({ type: String })
+  label: string;
+
+  @property({ type: String })
+  placeholder: string;
+
+  @property()
+  isOpen = false;
+
+  @property({ type: String })
+  value = '';
+
+  @provide({ context: dropdownContext })
+  dropdownContext: DropdownContext = {
+    value: this.value,
+    updateValue: (value) => {
+      this.value = value;
+      this._selectedOptionLabel = this.findLabelFromValue(value) ?? '';
+      this.triggerValueChangeEvent(value);
+    },
+  };
+
+  @state()
+  private _selectedOptionLabel = '';
+
+  render() {
+    return html`
+      <label>
+        ${this.label && html`<span>${this.label}</span>`}
+        <div class="input">
+          <input
+            @focus=${this.handleOpen}
+            value=${this._selectedOptionLabel}
+            placeholder=${this.placeholder}
+          />
+          <div class=${classMap({ icon: true, open: this.isOpen })} .innerHTML=${arrowDownIcon}></div>
+        </div>
+      </label>
+      ${this.isOpen
+        ? html`<div class="options">
+            <slot></slot>
+          </div>`
+        : ''}
+    `;
+  }
+
+  connectedCallback(): void {
+    super.connectedCallback();
+    document.addEventListener('click', this.closeOnOutsideClick);
+  }
+
+  disconnectedCallback(): void {
+    super.disconnectedCallback();
+    document.removeEventListener('click', this.closeOnOutsideClick);
+  }
+
+  private handleOpen() {
+    this.isOpen = true;
+  }
```

Comment:
Tiny thing, but we could consider to use arrow-function syntax for this guy as well (or go for the "regular" function syntax, as it looks better when we utilize life cycle hooks like `connectedCallback()`)